corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -51,10 +51,13 @@ def get_location_choices(query_context):
     # todo: consider making this an extensions framework similar to custom expressions
     # todo: does this need fancier permission restrictions and what not?
     # see e.g. locations.views.child_locations_for_select2
-    locations = SQLLocation.objects.filter(domain=query_context.domain)
     if query_context.query:
-        locations = locations.filter(name__icontains=query_context.query)
+        locations = SQLLocation.objects.filter_path_by_user_input(
+            domain=query_context.domain, user_input=query_context.query
+        )
+    else:
+        locations = SQLLocation.objects.filter(domain=query_context.domain)
     return [
-        Choice(loc.location_id, loc.name) for loc in
+        Choice(loc.location_id, loc.display_name) for loc in
```

Comment:
I decided not to use `get_path_display` since it looked like it would hit the DB once per location and also because it was pretty cluttered in the UI. I think we could add rendering options down the road after you convert to class-based choice providers, but my sense is that it's not worth prioritizing too much until someone asks for it concretely. yagni and all.